### PR TITLE
Support configuring java runtime from configmap or secret (env.valueFrom)

### PR DESCRIPTION
**Pull request by pavolloffay**
**Description:** <Describe what has changed.>

Supersedes #3373

This PR is much simpler it does not require fetching configmaps/secrets. 

If the `JAVA_TOOL_OPTIONS` is set (e.g. from configmap the operator sets another `JAVA_TOOL_OPTIONS=$(JAVA_TOOL_OPTIONS) -javaagent`

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #1814 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -1,3 +1,8 @@
 resources:
 - manager.yaml
-
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+images:
+- name: controller
+  newName: docker.io/pavolloffay/opentelemetry-operator
```
Comment (jaronoff97) [inline config/manager/kustomization.yaml:7]: should remove

File: tests/e2e-instrumentation/instrumentation-java/01-assert.yaml [hunk 2]
```diff
@@ -17,14 +17,19 @@ spec:
...
+          key: system-properties
     - name: OTEL_JAVAAGENT_DEBUG
       value: "true"
     - name: OTEL_INSTRUMENTATION_JDBC_ENABLED
       value: "false"
     - name: SPLUNK_PROFILER_ENABLED
       value: "false"
     - name: JAVA_TOOL_OPTIONS
-      value: ' -javaagent:/otel-auto-instrumentation-java/javaagent.jar'
+      value: '$(JAVA_TOOL_OPTIONS) -javaagent:/otel-auto-instrumentation-java/javaagent.jar'
```
Comment (jaronoff97) [inline tests/e2e-instrumentation/instrumentation-java/01-assert.yaml:32]: are we able to verify that this works? I didn't think that additive variables with the same name work?

Comment (jaronoff97) [reply tests/e2e-instrumentation/instrumentation-java/01-assert.yaml:32]: (verify in the test)

Comment (pavolloffay) [reply tests/e2e-instrumentation/instrumentation-java/01-assert.yaml:32]: it works, we are even using it for getting the pod name and namespace to resource attributes env var

Comment (pavolloffay) [reply tests/e2e-instrumentation/instrumentation-java/01-assert.yaml:32]: https://kubernetes.io/docs/tasks/inject-data-application/define-interdependent-environment-variables/

Comment (swiatekm) [reply tests/e2e-instrumentation/instrumentation-java/01-assert.yaml:32]: I think Jacob was asking if defining the same variable twice works as one would intuitively expect here.

Comment (pavolloffay) [reply tests/e2e-instrumentation/instrumentation-java/01-assert.yaml:32]: I have added the link into the source code. 

As I mention we already use it for other env vars

**Comment by pavolloffay** [comment]
PR updated

**Review by swiatekm — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**